### PR TITLE
api auth middleware

### PR DESCRIPTION
**Pull request by kyleferguson**
Wraps the `/api/v1` endpoints.

### CODE REVIEW

File: auth/guard.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	"net/http"
+	"regexp"
+)
+
+// APIMiddleware authenticates API requests
+func APIMiddleware(rw http.ResponseWriter, request *http.Request, next http.HandlerFunc) {
+	authHeader := request.Header.Get("Authorization")
+	if authHeader == "" {
+		rw.WriteHeader(401)
+		rw.Write([]byte("not authorized"))
```
Comment (jlaswell) [inline auth/guard.go]: I know this is splitting hairs, but this should be `Unauthorized`

File: auth/guard.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	if !validateAuthHeader(authHeader) {
+		log.Println("Authorization header is invalid:", authHeader)
+		rw.WriteHeader(401)
+		rw.Write([]byte("invalid authorization header"))
+		return
+	}
+
+	if !validateJWTToken(authHeader) {
+		rw.WriteHeader(401)
+		rw.Write([]byte("not authorized"))
```
Comment (jlaswell) [inline auth/guard.go:27]: Here as well

Comment (kyleferguson) [reply auth/guard.go:27]: Just a temporary response right now. API endpoints should return standardized JSON once we get there. Creating a task for it

File: auth/guard_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+	//"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+// 401 if no Authorization header is present
+func TestAPIMiddlewareNoHeader(t *testing.T) {
+	rw := new(responseWriterMock)
+	request, _ := http.NewRequest("GET", "/api/v1/foo", nil)
+	rw.On("Write", []byte("not authorized")).Once()
```
Comment (jlaswell) [inline auth/guard_test.go]: Here as well

**Review by jlaswell — APPROVED** [review]
Nice and simple 👍

**Pull request merged**